src/BoundaryConditions/fill_halo_regions.jl

```diff
@@ -56,60 +51,91 @@ function fill_halo_regions!(c::MaybeTupledData, boundary_conditions, indices, lo
     halo_tuple  = permute_boundary_conditions(boundary_conditions)
 
     # Fill halo in the three permuted directions (1, 2, and 3), making sure dependencies are fulfilled
-    for task in 1:3
+    for task in 1:length(halo_tuple[1])
```

Comment:
What's `length(halo_tuple[1])`?

Might be nice to write this a little differently:

```julia
meaningful_name_for_number_of_tasks = length(halo_tuple[1])
for task in 1:meaningful_name_for_number_of_tasks0
```

where `meaningful_name_for_number_of_tasks` describes the content of `length(halo_tuple[1])`.